### PR TITLE
Add translation support without Unicode

### PR DESCRIPTION
**Pull request by ihhub**
works on SDL 1 as well and only for game versions with correct language
alphabet

### CODE REVIEW

File: src/fheroes2/game/fheroes2.cpp [hunk 1]
```diff
@@ -116,25 +116,23 @@ namespace
...
     void SetLangEnvPath( const Settings & conf )
     {
-#ifdef WITH_TTF
-        if ( conf.Unicode() ) {
+        if ( !conf.ForceLang().empty() ) {
             System::SetLocale( LC_ALL, "" );
             System::SetLocale( LC_NUMERIC, "C" );
 
-            std::string mofile = conf.ForceLang().empty() ? System::GetMessageLocale( 1 ).append( ".mo" ) : std::string( conf.ForceLang() ).append( ".mo" );
+            const std::string mofile = conf.ForceLang().empty() ? System::GetMessageLocale( 1 ).append( ".mo" ) : std::string( conf.ForceLang() ).append( ".mo" );
```
Comment (oleg-derevenetz) [inline src/fheroes2/game/fheroes2.cpp:123]: I believe that `conf.ForceLang()` can't be empty here.

File: src/fheroes2/game/game_io.cpp [hunk 2]
```diff
@@ -230,11 +230,12 @@ fheroes2::GameMode Game::Load( const std::string & fn )
...
-    if ( conf.loadedFileLanguage() != "en" && conf.loadedFileLanguage() != conf.ForceLang() && !conf.Unicode() ) {
-        std::string warningMessage( "This is an saved game is localized for lang = " );
+    if ( !conf.loadedFileLanguage().empty() && conf.loadedFileLanguage() != "en" && conf.loadedFileLanguage() != conf.ForceLang() ) {
+        std::string warningMessage( "This is a saved game is localized for '" );
         warningMessage.append( conf.loadedFileLanguage() );
-        warningMessage.append( ", and most of the messages will be displayed incorrectly.\n \n" );
-        warningMessage.append( "(tip: set unicode = on)" );
+        warningMessage.append( "' language while current language is '" );
+        warningMessage.append( conf.ForceLang() );
+        warningMessage += '\'';
```
Comment (oleg-derevenetz) [inline src/fheroes2/game/game_io.cpp:238]: I believe better wording may be like this: "This saved game is localized to %s language, but the current language of the game is %s."

**Review by oleg-derevenetz — CHANGES_REQUESTED** [review]
Hi @ihhub I just left a few comments here, could you please take a look when you have time? Also I have the following questions:

1. Is there any value in the current "Unicode" implementation? As I can see, it doesn't even know how to work correctly with Unicode fonts;
2. As I can see, there was an attempt to bind in-game language to the system language (I mean, before, now it seems that this is regulated via config file only). I believe that by default (on empty `lang`) in-game language should be bound to the language of resources in the `DATA` folder (if possible, of course). Otherwise we may get weird mixes, when, for example, in-game language is Russian, but game resources are in English, so user will need to edit the config file to set `lang` by hand.

**Comment by ihhub** [comment]
> Hi @ihhub I just left a few comments here, could you please take a look when you have time? Also I have the following questions:
> 
> 1. Is there any value in the current "Unicode" implementation? As I can see, it doesn't even know how to work correctly with Unicode fonts;
> 2. As I can see, there was an attempt to bind in-game language to the system language (I mean, before, now it seems that this is regulated via config file only). I believe that by default (on empty `lang`) in-game language should be bound to the language of resources in the `DATA` folder (if possible, of course). Otherwise we may get weird mixes, when, for example, in-game language is Russian, but game resources are in English, so user will need to edit the config file to set `lang` by hand.

Hi @oleg-derevenetz ,
1. Unicode related code was in the old version of the project before 0.8.1 when we introduced current rendering engine. Before that rendering was based on SDL functions rather than on technology-separate logic leading to possibility to use SDL TTF support to convert fonts into SDL images for the subsequent rendering. However, such approach makes the game to have incorrect fonts. Maybe for some people it's not the issue but for others it's a big deal. So right now unicode related code has no value at all as we use in-game fonts. We can remove it.
2. This is a problem which I haven't figured out yet. Somehow we need to detect which language the original game resources contains. Whether it is French, Russian, Polish, German or English (the rest of languages were not officially supported). One way to do this is to verify almost every letter presented in alphabet to match the language. Let's take Russian language since you mentioned it. Russian language is Cyrillic alphabetic based where some letters are very similar to Latin but other aren't. Then we take Polish alphabet which is very similar to Latin. Without storing assets from the original game (we can't do this) how we detect that image in alphabet at place let's say 50 is H (russian letter) insted of K (latin letter).

Possible solutions for this problem:
- remember CRC32 from AGG files of the original copies of the game. Disadvantage: are we breaking the license by this?
- find alphabets which have letters on certain positions while others don't. By empty I mean an image bigger than 1x1 pixel. Some images are really 1x1 because they aren't used in the language. Disadvantage: 2 alphabets can have similar places which aren't empty.
- find an indicator: file, line and etc. which states the language of assets but I don't know about any of those.

File: src/fheroes2/game/game_io.cpp (hunk 2 above)
Comment (ihhub) [reply src/fheroes2/game/game_io.cpp:238]: Got it. I'll fix it. Thank you!

File: src/fheroes2/game/fheroes2.cpp (hunk 1 above)
Comment (ihhub) [reply src/fheroes2/game/fheroes2.cpp:123]: Good catch!

**Comment by oleg-derevenetz** [comment]
Hi @ihhub

> So right now unicode related code has no value at all as we use in-game fonts. We can remove it.

I can do it in a separate PR after merging this one.

> remember CRC32 from AGG files of the original copies of the game. Disadvantage: are we breaking the license by this?

I don't really think that we are breaking the license by providing a hash sum for certain data files. If this were the case, many antivirus software or package managers would "break the license" in this way :)

**Review by oleg-derevenetz — APPROVED** [review]

**Comment by undef21** [comment]
> remember CRC32 from AGG files of the original copies of the game. Disadvantage: are we breaking the license by this?

Such project as ScummVM [contains](https://github.com/scummvm/scummvm/blob/master/engines/agos/detection_tables.h) md5 sums of supported game resources for many years.

**Pull request merged**